src/test/java/org/elastos/did/VerifiableCredentialTest.java

```diff
@@ -167,4 +281,721 @@ public void testParseAndSerializeJsonCredential(int version, String did, String
+		DIDURL signKey = null;
+		if (doc.getControllerCount() > 1) {
+			Random rnd = new Random();
+			int index = (rnd.nextInt() & Integer.MAX_VALUE) % doc.getControllerCount();
+			signKey = doc.getControllers().get(index).resolve().getDefaultPublicKeyId();
+		}
+
+		credential.declare(signKey, TestConfig.storePass);
+
+		DIDURL id = credential.getId();
+		VerifiableCredential resolved = VerifiableCredential.resolve(id);
+		assertNotNull(resolved);
+
+		assertEquals(credential.toString(), resolved.toString());
+
+		CredentialMetadata metadata = resolved.getMetadata();
+		assertNotNull(metadata);
+		assertNotNull(metadata.getPublished());
+		assertNotNull(metadata.getTransactionId());
+		assertFalse(resolved.isRevoked());
+
+		assertTrue(credential.wasDeclared());
+
+		credential.revoke(signKey, TestConfig.storePass);
+
+		resolved = VerifiableCredential.resolve(id);
+		assertNotNull(resolved);
+
+		assertEquals(credential.toString(), resolved.toString());
+
+		metadata = resolved.getMetadata();
+		assertNotNull(metadata);
+		assertNotNull(metadata.getPublished());
+		assertNotNull(metadata.getTransactionId());
+		assertTrue(resolved.isRevoked());
+
+		CredentialBiography bio = VerifiableCredential.resolveBiography(id, credential.getIssuer());
+		assertNotNull(bio);
+		assertEquals(2, bio.getAllTransactions().size());
+		assertEquals(IDChainRequest.Operation.REVOKE, bio.getTransaction(0).getRequest().getOperation());
+		assertEquals(IDChainRequest.Operation.DECLARE, bio.getTransaction(1).getRequest().getOperation());
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+    	"2,foobar,license",
+    	"2,foobar,services",
+    	"2,foo,email"})
+    public void testRevokeCrendentialWithDifferentKey(int version, String did, String vc)
+			throws DIDException, IOException {
+	   	TestData.CompatibleData cd = testData.getCompatibleData(version);
+	   	cd.loadAll();
+
+		VerifiableCredential credential = cd.getCredential(did, vc);
+		assertFalse(credential.wasDeclared());
+
+		// Sign key for customized DID
+		DIDDocument doc = credential.getSubject().getId().resolve();
+		DIDURL signKey = null;
+		int index = 0;
+		if (doc.getControllerCount() > 1) {
+			Random rnd = new Random();
+			index = (rnd.nextInt() & Integer.MAX_VALUE) % doc.getControllerCount();
+			signKey = doc.getControllers().get(index).resolve().getDefaultPublicKeyId();
+		}
+
+		credential.declare(signKey, TestConfig.storePass);
+
+		DIDURL id = credential.getId();
+		VerifiableCredential resolved = VerifiableCredential.resolve(id);
+		assertNotNull(resolved);
+
+		assertEquals(credential.toString(), resolved.toString());
+
+		CredentialMetadata metadata = resolved.getMetadata();
+		assertNotNull(metadata);
+		assertNotNull(metadata.getPublished());
+		assertNotNull(metadata.getTransactionId());
+		assertFalse(resolved.isRevoked());
+
+		assertTrue(credential.wasDeclared());
+
+		if (doc.getControllerCount() > 1) {
+			index = ++index % doc.getControllerCount();
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [DLS: Overwritten increment (DLS_OVERWRITTEN_INCREMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dls-overwritten-increment-dls-overwritten-increment).
The code performs an increment operation (e.g., i++) and then immediately overwrites it. For example, i = i++ immediately overwrites the incremented value with the original value.